src/averageAges.js

```diff
@@ -37,7 +40,14 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const womenAverageAge = people
+    .filter(person =>
+      withChildren
+        ? people.some(child => child.mother === person.name)
```

Comment:
It's a good solution. Just for you information, you could do it without the nested loop